src/webots/gui/WbGuiApplication.cpp

```diff
@@ -193,110 +194,164 @@ void WbGuiApplication::parseStreamArguments(const QString &streamArguments) {
+  parser.addOption(minimizeOption);
+  QCommandLineOption fullscreenOption("fullscreen", QCoreApplication::translate("main", "View window in fullscreen"));
+  parser.addOption(fullscreenOption);
+  QCommandLineOption modeOption("mode",
+                                QCoreApplication::translate("main", "Specify speed of simulation (default is realtime)"),
+                                QCoreApplication::translate("main", "speed"), "realtime");
+  parser.addOption(modeOption);
+  QCommandLineOption convertOption("convert", QCoreApplication::translate("main", "Convert."));
+  parser.addOption(convertOption);
+  QCommandLineOption noRenderingOption("no-rendering", QCoreApplication::translate("main", "No rendering"));
+  parser.addOption(noRenderingOption);
+  QCommandLineOption sysinfoOption("sysinfo", QCoreApplication::translate("main", "System information"));
+  parser.addOption(sysinfoOption);
+  QCommandLineOption batchOption("batch", QCoreApplication::translate("main", "Batch"));
+  parser.addOption(batchOption);
+  QCommandLineOption updateProtoCacheOption("update-proto-cache", QCoreApplication::translate("main", "Update the PROTO cache"),
+                                            QCoreApplication::translate("main", "task"));
+  parser.addOption(updateProtoCacheOption);
+  QCommandLineOption updateWorldOption("update-world", QCoreApplication::translate("main", "Update world"));
+  parser.addOption(updateWorldOption);
+  QCommandLineOption x3DMetaFileExportOption("enable-x3d-meta-file-export",
+                                             QCoreApplication::translate("main", "Enable x3d meta file export"));
+  parser.addOption(x3DMetaFileExportOption);
+  QCommandLineOption streamOption("stream", QCoreApplication::translate("main", "Stream"));
+  parser.addOption(streamOption);
+  QCommandLineOption stdoutOption("stdout", QCoreApplication::translate("main", "stdout"));
+  parser.addOption(stdoutOption);
+  QCommandLineOption stderrOption("stderr", QCoreApplication::translate("main", "stderr"));
+  parser.addOption(stderrOption);
+  QCommandLineOption logPerfomanceOption("log-performance", QCoreApplication::translate("main", "Log performance"),
+                                         QCoreApplication::translate("main", "logArgument"));
+  parser.addOption(logPerfomanceOption);
+  parser.addOptions({
+    {"disable-logging", QCoreApplication::translate("main", "Disable logging")},
+    {"enable-logging", QCoreApplication::translate("main", "Enable logging")},
+    {"log-level", QCoreApplication::translate("main", "Log level")},
+    {"no-sandbox", QCoreApplication::translate("main", "No sandbox")},
+    {"single-process", QCoreApplication::translate("main", "Single Process")},
+    {"remote-debugging-port", QCoreApplication::translate("main", "Remote Debugging Port")},
+  });
+
+  parser.process(QCoreApplication::arguments());
+  parser.parse(QCoreApplication::arguments());
```

Comment:
Maybe you should check the return value of the `parse()` method and if false, you should print on stderr the error, which you can retrieve with the `errorText()` method. See example [here](https://doc.qt.io/qt-6/qcommandlineparser.html).